lib/routemaster/faraday_middleware/caching.rb

```diff
@@ -0,0 +1,46 @@
+require 'routemaster/cache'
+
+module Routemaster::FaradayMiddleware
+  class Caching
+
+    def initialize(app, cache = Routemaster::Cache.new)
+      @app = app
+      @cache = cache
+    end
+
+    def call(env)
+      return @app.call(env) unless env.method == :get
+
+      url, version, locale = url(env), version(env), locale(env)
```

Comment:
Sorry, are you talking about the parallel assignment, or method calls? Parallel assignment actually seems to be faster on 2.3 https://github.com/JuanitoFatas/fast-ruby/blob/master/code/general/assignment.rb

I can split it out if you think it is hard to read.